CMakeLists.txt

```diff
@@ -377,18 +383,37 @@ endif()
 # use -DBUILD_SHARED_LIBS=ON|OFF to control static/shared
 add_library(arpack ${arpackutil_STAT_SRCS} ${arpacksrc_STAT_SRCS} ${arpacksrc_ICB})
 
-target_link_libraries(arpack BLAS::BLAS)
-target_link_libraries(arpack LAPACK::LAPACK ${EXTRA_LDFLAGS})
+target_link_libraries(arpack
+  PUBLIC
+  $<INSTALL_INTERFACE:$<IF:$<BOOL:${BUILD_SHARED_LIBS}>,,BLAS::BLAS>>
+  $<INSTALL_INTERFACE:$<IF:$<BOOL:${BUILD_SHARED_LIBS}>,,LAPACK::LAPACK>>
+  $<BUILD_INTERFACE:BLAS::BLAS>
```

Comment:
@juanjosegarciaripoll: could you just explain what you do here? difference between  INSTALL_INTERFACE and BUILD_INTERFACE? (I try to improve my knowldge of cmake ! :))

Reply:
BUILD_INTERFACE macros are replaced when the library and its components (e.g. tests) are compiled. In this case, we are declaring that the library depends on BLAS::BLAS, LAPACK::LAPACK and that the components that are built with it (the tests) will get this dependency automatically transferred to them (PUBLIC).

INSTALL_INTERFACE is a macro that is only expanded when the library is finally exported, in this case to an arpacksTarget.cmake file. Through this text we make the library export its dependency on BLAS and LAPACK only when it is built as a statically linked library. I believe this is the minimum that must be done.

Reply:
Thanks for clarifications!